Fix oriented beam query.

feuerste opened the pull request.

nnmm commented:
I am a bit worried about Aabb being transformed into a larger Aabb. Would you also be fine with leaving src/iterator.rs and src/math.rs as they are for now? Or with panicking instead?

nnmm commented on src/math.rs:
```diff
@@ -42,14 +42,15 @@ where
     // TODO(catevita): return Relation
     fn intersects_aabb3(&self, aabb: &Aabb3<S>) -> bool;
     fn transformed(&self, global_from_query: &Isometry3<S>) -> Box<dyn PointCulling<S>>;
+    fn corners(&self) -> [Point3<S>; 8];
```
Wouldn't this be a little limiting in the future to have this in the trait?

feuerste replied:
I currently think of culling as a bounding volume, which always has corners. I was even thinking about removing `AllPoints` from it, but it would have been too much effort in this PR.
If we need more than a bounding volume, e.g. sphere in the future, we can still take it out of the trait.

feuerste replied:
It also nicely integrates with the changes in the s2 cells mod below...

feuerste commented:
> I am a bit worried about Aabb being transformed into a larger Aabb. Would you also be fine with leaving src/iterator.rs and src/math.rs as they are for now? Or with panicking instead/at least printing a warning?

As discussed offline, I reverted it to being transformed into an Obb. We also removed the oriented beam query altogether, to get rid of its hackish implementations.

feuerste commented:
Thanks for your review!
